apps/firebase/src/celo-adapter.ts

```diff
@@ -33,26 +54,44 @@ export class CeloAdapter {
    *
    */
   async convertExtraStablesToCelo(amount: string) {
+    const mento = this.mento
+    if (!mento) {
+      throw new Error("Must call init() first")
+    }
+    const celoContractAddress = await this.kit.celoTokens.getAddress(Token.CELO)
     await this.kit.celoTokens.forStableCeloToken(async (info) => {
       try {
-        console.log('converting', info.symbol)
         const stableToken = await this.kit.celoTokens.getWrapper(info.symbol as StableToken)
         const faucetBalance = await stableToken.balanceOf(this.defaultAddress)
         const MIN_BALANCE_IN_WEI = "25000000000000000000000" // 25K
         if (faucetBalance.isLessThan(new BigNumber(MIN_BALANCE_IN_WEI))) {
           console.log('skipping', info.symbol, faucetBalance.toString())
           return
         }
-        const exchangeContract = await this.kit.contracts.getContract(info.exchangeContract)
-
-        const [quote] = await Promise.all([
-          exchangeContract.quoteStableSell(amount),
-          stableToken.increaseAllowance(exchangeContract.address, amount).sendAndWaitForReceipt()
-        ]);
-
+        console.log('converting', info.symbol)
 
-        const tx = await exchangeContract.sellStable(amount, quote.multipliedBy(0.99).integerValue(BigNumber.ROUND_UP))
-        await tx.sendAndWaitForReceipt()
+        const allowanceTxObj = await mento.increaseTradingAllowance(
+          stableToken.address,
+          amount
+        )
+
+        const allowanceTx = await this.signer.sendTransaction(allowanceTxObj);
+        await allowanceTx.wait();
+
+        const quoteAmountOut = await mento.getAmountOut(
+          stableToken.address,
+          celoContractAddress,
+          amount
+        );
+        const expectedAmountOut = quoteAmountOut.mul(99).div(100); // allow 1% slippage from quote
```

Comment:
Could you just do `.mul(0.99)`?

Reply:
i think so. at least thats how it was before. 

to be frank i copied this part from the mento example app

Reply:
answer is no. it results in error